Presentation: Content customization doc enhancements (#3078)
iTwin/itwinjs-core: docs/presentation/Content/CalculatedPropertiesSpecification.md

# Calculated Properties Specification

> TypeScript type: [CalculatedPropertiesSpecification]($presentation-common).

This content modifier allows including additional calculated properties into the content.

## Attributes

| Name                              | Required? | Type                                         | Default |
| --------------------------------- | --------- | -------------------------------------------- | ------- |
| [`label`](#attribute-label)       | Yes       | `string`                                     |         |
| [`value`](#attribute-value)       | Yes       | [ECExpression](../Advanced/ECExpressions.md) |         |
| [`priority`](#attribute-priority) | No        | `number`                                     | `1000`  |

### Attribute: `label`

Specifies label of the calculated property. Supports [localization](../Advanced/Localization.md).

```ts
[[include:Content.Customization.CalculatedPropertiesSpecification.Label.Ruleset]]
```

![Example of using "label" attribute](./media/calculatedpropertiesspecification-with-label-attribute.png)

### Attribute: `value`

> Defines an expression to calculate the value. The expression can use [ECInstance](../Advanced/ECExpressions.md#ecinstance)
> and [Ruleset Variables](../Advanced/ECExpressions.md#ruleset-variables-user-settings) symbol contexts.

Reviewer: I think it makes more sense to link the main entry point to ECExpressions documentation, unless there is something special to mention about this attribute.

```suggestion
Defines an [ECExpression](../Advanced/ECExpressions.md) that computes the property value.
```

Author: Different attributes support different ECExpression symbols. I think it's important to mention which ones can be used in this specific attribute.

```ts
[[include:Content.Customization.CalculatedPropertiesSpecification.Value.Ruleset]]
```

![Example of using "value" attribute](./media/calculatedpropertiesspecification-with-value-attribute.png)

### Attribute: `priority`

> **Default value:** `1000`

Assign a custom [Field.priority]($presentation-common) to the property. It's up to the UI component to make sure that priority
is respected - properties with higher priority should appear before or above properties with lower priority.

> 

Reviewer: Am I interpreting this right: instance properties are not ordered by `priority` in the content response? Is this the point you were trying to get across?

Author: To be exact, priority is assigned to a `Field` which is created for one or more properties. We only assign priority to them and don't do any kind of sorting on the backend - that should be done by the component displaying the results.
